File: C++/observer/simple/SimpleObserver.cpp
//
// Created by Chans on 2020/2/5.
//
#include <iostream>
#include "SimpleObserver.h"

SimpleObserver::SimpleObserver(Subject *simpleSubject) {
    subject = simpleSubject;
    simpleSubject->registerObserver(this);
}

void SimpleObserver::update(int value) {
    this->value = value;
    display();
}

void SimpleObserver::display() {
    std::cout << "Value: " << value << std::endl;
}

File: Kotlin/Obs/src/simple/SimpleObserver.kt
package simple

class SimpleObserver(observable: Observable): Observer {
    private var value: Int = 0
    init {
        observable.registerObserver(this)
    }

    override fun update(value: Int) {
        this.value = value
        display()
    }

    fun display(){
        println("Value: " + value)
    }
}